protocol/message_validator.go

```diff
@@ -364,7 +364,7 @@ func ValidateReceivedEmojiReaction(emoji *protobuf.EmojiReaction, whisperTimesta
 		return errors.New("chat-id can't be empty")
 	}
 
-	if emoji.Type == protobuf.EmojiReaction_UNKNOWN_EMOJI_REACTION_TYPE {
+	if emoji.Type == protobuf.EmojiReaction_UNKNOWN_EMOJI_REACTION_TYPE && emoji.Emoji == "" {
```

Comment:
Maybe it should be a bit smarter?
Smth logically equal to `EMOJI_V1 || EMOJI_V2`

Reply:
Are you suggesting that I add a Version property to the protobuf? Just making sure I understand.

Reply:
@jrainville not really. 

Something like this:
```suggestion
	if (emoji.Type == protobuf.EmojiReaction_UNKNOWN_EMOJI_REACTION_TYPE) != (emoji.Emoji == "")
```

Meatning that it should always be one of:
- `emoji.Type != Unknown && emoji.Emoji == ""` (version 1)
- `emoji.Type == Unknown && emoji.Emoji != ""` (version 2)
... but never both.

Reply:
Good suggestion. Done. FYI, it needed to be `==` in the middle :wink:

Reply:
Noice, `==` indeed 😄 

Can you pleeease implement a simple unit test for this?

Reply:
Ok, so I had to undo the change you proposed @igor-sirotin because it broke the functional tests. The reason is that if the current app uses the old API, then it will send both the `emojiID` and the `emoji`. So the condition wouldn't pass.

My first version makes sure that we always have a valid emoji, whatever version is used.

I update the functional test too to validate that V1 and V2 both work (though I will remove V1 soon once everything is in place in the client)

Reply:
I just realized we have a unit test already for ValidateReceivedEmojiReaction so I added two new cases to validate the new condition